src/main/java/hillel/spring/doctors/DoctorRepository.java

```diff
@@ -0,0 +1,29 @@
+package hillel.spring.doctors;
+
+
+import hillel.spring.doctors.model.Doctor;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.Collection;
+import java.util.List;
+
+@Repository
+public interface DoctorRepository extends JpaRepository<Doctor, Integer> {
+
+    List<Doctor> findBySpecializationAndNameStartsWithIgnoreCase(@Param("specialization") String specialization, @Param("name") String name);
+
+    List<Doctor> findBySpecialization(@Param("specialization") String specialization);
+
+    List<Doctor> findByNameStartsWithIgnoreCase(@Param("name") String name);
+
+    List<Doctor> findAllBySpecializationIn(@Param("specializations") List<String> specializations);
```

Comment:
The `@Param` annotation is needed for custom queries only (in other words for queries in `@Query` annotation)